nntrainer/app_context.h

```diff
@@ -94,8 +94,16 @@ class AppContext {
    * @throws std::invalid_parameter if library_path is invalid or library is
    * invalid
    */
-  int registerLayerPlugin(const std::string &library_path,
-                          const std::string &base_path = "");
+  int registerLayer(const std::string &library_path,
+                    const std::string &base_path = "");
+
+  /**
+   * @brief register Layer from a directory.
+   *
+   * @param base_path a directory path to search layer's
+   * @return std::vector<int> list of integer key to create a layer
```

Comment:
add a note here for the expected behavior when the same layer is registered twice.

Reply:
you might also consider adding a test case for the same.

Reply:
It is responsibility of `registerFactory` and there is a testcase for that: `unittest_nntrainer_appcontext.cpp:RegisterFactoryWithClashingKey_n`

I'll add a note comment.

Reply:
Then please add a comment to ensure that NNTRAINER_PATH cannot contain duplicate paths.
Also, if we don't support multiple paths in NNTRAINER_PATH, add as a comment as well.

Reply:
Good point, I'll add

Reply:
Added to the place where NNTRAINER_PATH is explained